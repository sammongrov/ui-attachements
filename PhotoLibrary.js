import React, { Component } from 'react';
import { View, TouchableOpacity, Dimensions } from 'react-native';
import { styles } from 'react-native-theme';
import CameraRollPicker from 'react-native-camera-roll-picker';
import { Colors } from '@ui/theme_default';
import { Icon, NavBar, Screen } from '@ui/components';
import { Actions } from 'react-native-router-flux';

export default class PhotoLibrary extends Component {
  state = {
    layout: {
      width: Dimensions.get('window').width,
    },
  };

  _onLayout = (event) => {
    this.setState({
      layout: {
        width: event.nativeEvent.layout.width,
      },
    });
  };

  render() {
    const { layout } = this.state;
    return (
      <Screen onLayout={this._onLayout}>
        <View style={[styles.flex1, { width: layout.width, backgroundColor: '#000' }]}>
          <NavBar
            leftComponent={
              <TouchableOpacity
                style={[styles.navSideButtonDimension, styles.alignJustifyCenter]}
                onPress={() => {
                  if (Actions.currentScene === 'PhotoLibraryScene') {
                    Actions.pop();
                  }
                }}
              >
                <Icon
                  name="chevron-left"
                  type="material-community"
                  color={Colors.NAV_ICON}
                  size={36}
                />
              </TouchableOpacity>
            }
            titleText="Image Gallery"
          />
          <CameraRollPicker
            maximum={1}
            assetType="All"
            pageSize={2}
            imagesPerRow={3}
            callback={(images, dataToUpload) => {
              if (Actions.currentScene === 'PhotoLibraryScene') {
                Actions.ImagePreview({
                  imageUrl: dataToUpload.uri,
                  onSuccessAction: (imageCaption) => {
                    Actions.popTo('ChatRoomScene');
                    setTimeout(() => {
                      Actions.refresh({
                        dataToUpload,
                        imageCaption,
                      });
                    }, 0);
                  },
                });
              }
              if (Actions.currentScene === 'ReplyPhotoLibraryScene') {
                Actions.ImagePreview({
                  imageUrl: dataToUpload.uri,
                  onSuccessAction: (imageCaption) => {
                    Actions.popTo('ReplyMessageScene');
                    setTimeout(() => {
                      Actions.refresh({
                        dataToUpload,
                        imageCaption,
                      });
                    }, 0);
                  },
                });
              }
            }}
            selected={[]}
            containerWidth={layout.width}
            selectedMarker={null}
          />
        </View>
      </Screen>
    );
  }
}
