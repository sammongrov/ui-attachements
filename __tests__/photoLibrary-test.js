import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
/* eslint-disable */
import { Dimensions } from 'react-native';
import CameraRollPicker from 'react-native-camera-roll-picker';
/* eslint-enable */
import Adapter from 'enzyme-adapter-react-16';
import { Actions } from 'react-native-router-flux';
import PhotoLibrary from '../PhotoLibrary';

configure({ adapter: new Adapter() });
// const onConfirmServer = jest.fn();

jest.mock('react-native-camera-roll-picker', () => {
  /* eslint-disable */
  const React = require('React');
  const PropTypes = require('prop-types');
  return class MockCameraRollPicker extends React.Component {
    static propTypes = { children: PropTypes.any };

    static defaultProps = { children: '' };

    render() {
      return React.createElement('CameraRollPicker', this.props, this.props.children);
    }
    /* eslint-enable */
  };
});

jest.mock('react-native-router-flux', () => ({
  Actions: {
    ImagePreview: jest.fn((data) => {
      data.onSuccessAction();
    }),
    popTo: jest.fn(),
    refresh: jest.fn(),
    currentScene: 'PhotoLibraryScene',
    pop: jest.fn(),
  },
}));

jest.mock('Dimensions', () => ({
  get: () => ({ width: 720, height: 360 }),
}));

jest.useFakeTimers();

it('render correctly with props', () => {
  const tree = renderer.create(<PhotoLibrary />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('initial state is set', () => {
  const root = shallow(<PhotoLibrary />);
  expect(root.state().layout.width).toBe(720);
});

it('Navbar onPress calls Actions.pop', () => {
  Actions.currentScene = 'PhotoLibraryScene';
  const root = shallow(<PhotoLibrary />);
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity');
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(1);
});

it('Navbar onPress does not calls Actions.pop', () => {
  Actions.currentScene = 'ChatList';
  Actions.pop.mockClear();
  const root = shallow(<PhotoLibrary />);
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity');
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(0);
});

it('photolibrary, CameraRollPicker first if case', () => {
  const dataToUpload = {
    uri: '/image/image-1.jpg',
  };
  const images = true;
  const imageCaption = 'my cool image';

  Actions.currentScene = 'PhotoLibraryScene';
  Actions.refresh.mockClear();
  Actions.popTo.mockClear();
  setTimeout.mockClear();
  Actions.ImagePreview = jest.fn((obj) => {
    obj.onSuccessAction(imageCaption);
  });

  const rootComponent = shallow(<PhotoLibrary />);
  const authorInputComponent = rootComponent.find({ assetType: 'All' });
  authorInputComponent.props().callback(images, dataToUpload);

  expect(Actions.ImagePreview).toBeCalled();
  expect(Actions.popTo).toBeCalledWith('ChatRoomScene');

  jest.runAllTimers();

  expect(Actions.refresh).toBeCalledWith({ dataToUpload, imageCaption });
});

it('photolibrary, CameraRollPicker second if case', () => {
  const dataToUpload = {
    uri: '/image/image-1.jpg',
  };
  const images = true;
  const imageCaption = 'my cool image';

  Actions.currentScene = 'ReplyPhotoLibraryScene';
  Actions.refresh.mockClear();
  Actions.popTo.mockClear();
  setTimeout.mockClear();
  Actions.ImagePreview = jest.fn((obj) => {
    obj.onSuccessAction(imageCaption);
  });

  const rootComponent = shallow(<PhotoLibrary />);
  const authorInputComponent = rootComponent.find({ assetType: 'All' });
  authorInputComponent.props().callback(images, dataToUpload);

  expect(Actions.ImagePreview).toBeCalled();
  expect(Actions.popTo).toBeCalledWith('ReplyMessageScene');

  jest.runAllTimers();

  expect(Actions.refresh).toBeCalledWith({ dataToUpload, imageCaption });
});

it('_onLayout method is called', () => {
  const event = { nativeEvent: { layout: { width: 360 } } };
  const root = shallow(<PhotoLibrary />);
  root
    .find('Screen')
    .props()
    .onLayout(event);
  expect(root.state().layout.width).toBe(360);
});
