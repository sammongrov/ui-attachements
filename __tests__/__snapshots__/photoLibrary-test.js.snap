// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly with props 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "backgroundColor": "#000",
              "width": 720,
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              null,
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#2E88FF",
                      "fontSize": 36,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                type="material-community"
              >
                ÔÖÅ
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                undefined,
                undefined,
                null,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "OpenSans-Regular",
                  },
                  Array [
                    undefined,
                    null,
                  ],
                ]
              }
            >
              Image Gallery
            </Text>
          </View>
          <View
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          />
        </View>
        <CameraRollPicker
          assetType="All"
          callback={[Function]}
          containerWidth={720}
          imagesPerRow={3}
          maximum={1}
          pageSize={2}
          selected={Array []}
          selectedMarker={null}
        >
          
        </CameraRollPicker>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
